enumerables.rb

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/ruby
+# frozen_string_literal: true
+
+# rubocop:disable ModuleLength
+
+module Enumerable
+  # my_each
+  def my_each
+    if block_given?
+      (0...size).each do |i|
+        yield(self[i])
+      end
+    else
+      to_enum
+    end
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    if block_given?
+      (0...size).each do |i|
+        yield(self[i], i)
+      end
+    else
+      to_enum
+    end
+  end
+
+  # my_select
+
+  def my_select
+    if block_given?
+      if instance_of?(Array)
+        temp_array = []
+        my_each do |item|
+          temp_array << item if yield(item)
+        end
+        temp_array
+      end
+    else
+      to_enum
+    end
+  end
+
+  # my_all?
+
+  def my_all?
+    if block_given?
+      temp_array = []
+      my_each do |item|
+        temp_array << yield(item)
+      end
+      (0...temp_array.size).each do |i|
+        return true if temp_array[i] == true
+
+        return false
+      end
+    else
+      to_enum
+    end
+  end
+
+  # my_any?
+
+  def my_any?
+    if block_given?
+      temp_array = []
+      my_each do |item|
+        temp_array << yield(item)
+      end
+      (0...temp_array.size).each do |i|
+        return true if temp_array[i] == true || temp_array.nil?
+      end
+    else
+      true
+    end
+  end
+
+  # my_none?
+
+  def my_none?
+    if block_given?
+      temp_array = []
+      my_each do |item|
+        temp_array << yield(item)
+      end
+      (0...temp_array.size).each do |i|
+        return true if temp_array[i] == false || temp_array.nil?
+      end
+    else
+      true
+    end
+  end
+
+  # my_count
+
+  def my_count
+    if instance_of?(Array)
+      temp_array = []
+      my_each do |item|
+        temp_array << item if yield(item)
+      end
+      temp_array.size
+    end
+    return unless block_given?
+  end
+
+  # my_map
+
+  # def my_map(&block)
+  #    if block_given?
+  #         temp_array = []
+  #         self.my_each do |item|
+  #             temp_array << block.call(item)
+  #         end
+  #         temp_array
+  #         else
+  #         self.to_enum
+  #     end
+  # end
+
+  # my_inject
+
+  def my_inject(param = nil)
+    if block_given?
+      if param.nil?
+        result = shift
+        my_each_with_index do |_, index|
+          result = yield(result, self[index])
+        end
+        return result
+      else
+        result = 0
+        i = param
+        (i...size).each do |n|
+          result = yield(result, self[n])
+        end
+      end
```

Comment:
Did you test for when it is called on a range? Remember, the Range class also includes the Enumerable method.

Also, what should happen when a block is not given? 

`[1,2,3,4,5].my_inject(:+)` should equal `15` not `nil`